rules_haskell_tests/MODULE.bazel

```diff
@@ -308,3 +293,71 @@ use_repo(
     rules_haskell_dependencies,
     "zlib",
 )
+
+haskell_toolchains = use_extension(
+    "@rules_haskell//extensions:haskell_toolchains.bzl",
+    "haskell_toolchains",
+)
+
+test_ghc_version = "9.2.5"
+
+test_ghcopts = [
+    "-XStandaloneDeriving",  # Flag used at compile time
+    "-threaded",  # Flag used at link time
+    # Used by `tests/repl-flags`
+    "-DTESTS_TOOLCHAIN_COMPILER_FLAGS",
+    # this is the default, so it does not harm other tests
+    "-XNoOverloadedStrings",
+]
+
+test_haddock_flags = ["-U"]
+
+test_repl_ghci_args = [
+    # The repl test will need this flag, but set by the local
+    # `repl_ghci_args`.
+    "-UTESTS_TOOLCHAIN_REPL_FLAGS",
+    # The repl test will need OverloadedString
+    "-XOverloadedStrings",
+]
+
+test_cabalopts = [
+    # Used by `tests/cabal-toolchain-flags`
+    "--ghc-option=-DTESTS_TOOLCHAIN_CABALOPTS",
+    "--haddock-option=--optghc=-DTESTS_TOOLCHAIN_CABALOPTS",
+]
+
+cabalopts_windows = test_cabalopts + [
+    # To avoid ghcide linking errors with heapsize on Windows of the form
+    #
+    #   unknown symbol `heap_view_closurePtrs'
+    #
+    # See https://github.com/haskell/ghcide/pull/954
+    "--disable-library-for-ghci",
+]
```

Comment:
> The configuration variables such as `test_repl_ghci_args` are duplicated inside the [MODULE.bazel](https://github.com/tweag/rules_haskell/pull/1914/files?diff=unified&w=0#diff-c385cb2f1e4a4c4652191639dc4ccb1c8e048d9ebc33d61b12dae27c57b45fccR302-R336) file because it is not possible to `load` them at the moment (see [bazelbuild/bazel#17880](https://github.com/bazelbuild/bazel/issues/17880)). For the same reason it is not possible to rely on `is_windows` to configure the `cabalops` variable so the windows toolchains are declared separately in this file.

Perhaps another approach could be for the module extension/tags to accept a label to a JSON file holding this configuration? (Either way, out of scope for this PR)